### PR TITLE
New randomization opcodes

### PR DESCRIPTION
**Pull request by MiranDMC**
added **pick_random_int**
added **pick_random_float**
added **pick_random_text**
added **random_chance**

### CODE REVIEW

File: cleo_plugins/Math/Math.cpp [hunk 1]
```diff
@@ -434,4 +446,96 @@ class Math
         OPCODE_CONDITION_RESULT(value != 0);
         return OR_CONTINUE;
     }
-} Math;
+
+    //2705=-1,pick_random_int values %d% store_to %d%
```
Comment (x87) [inline cleo_plugins/Math/Math.cpp:450]: picking random array element is called sampling

please rename these it `sample_int`, `sample_float`, `sample_string`

**Review by x87 — CHANGES_REQUESTED** [review]

File: cleo_plugins/Math/Math.cpp (hunk 1 above)
Comment (MiranDMC) [reply cleo_plugins/Math/Math.cpp:450]: Hm not sure about that. I would never guess it takes random element from set.
"Sample - get a representative experience of". Especially in software development it is already used for totally different meanings (like texture sampling).
Originally I considered also "select". Does some programming language have similar function?

Comment (MiranDMC) [reply cleo_plugins/Math/Math.cpp:450]: pick (noun) · picks (plural noun)
    an act or the right of selecting something from a number of alternatives

I was baffling about name already and this seem to be the best one. Other options seems like fancy ways to not say "pick".

**random_pick_int** maybe?

Comment (x87) [reply cleo_plugins/Math/Math.cpp:450]: sample is a verb too

Comment (x87) [reply cleo_plugins/Math/Math.cpp:450]: https://www.geeksforgeeks.org/python-random-sample-function/
https://lodash.com/docs/4.17.15#sample

Comment (MiranDMC) [reply cleo_plugins/Math/Math.cpp:450]: Yeah, like sampling input sensor by averaging input over sampling window time, or sampling audio to fill given time by repeating small portion. 
I think all randomization opcodes should have word "random" in name. Then searching in SBL actually provides user with options to select from.
**Math.Sample(1,2.3)** looks meaningless.

**Review by x87 — APPROVED** [review]

**Pull request merged**